### PR TITLE
OAuth2 migration

### PR DESCRIPTION
**Pull request by v3rm0n**
Migrate to the new Spring Security OAuth2 stack

### CODE REVIEW

File: src/test/groovy/ee/tuleva/onboarding/payment/PaymentIntegrationSpec.groovy [hunk 1]
```diff
@@ -115,9 +135,9 @@ class PaymentIntegrationSpec extends Specification {
     SecurityContextHolder.clearContext()
     expectThatPaymentCallbackRedirectsUser()
     expectThatPaymentCallbackCreatedOnePayment(aUser)
+    mockSecurityContext()
     mockEpisTransactionsForPayment()
     expectToBeAbleToReceivePaymentNotification()
```
Comment (ErkoRisthein) [inline src/test/groovy/ee/tuleva/onboarding/payment/PaymentIntegrationSpec.groovy:140]: In production, the payment notification incoming request will have no security context.

File: src/main/java/ee/tuleva/onboarding/aml/AmlAutoChecker.java [hunk 2]
```diff
@@ -45,14 +44,12 @@ public void beforeLogin(BeforeTokenGrantedEvent event) {
...
     Person person = event.getPerson();
-    String token = event.getAccessToken().getValue();
 
     userService
         .findByPersonalCode(person.getPersonalCode())
         .ifPresent(
             user -> {
-              ContactDetails contactDetails =
-                  contactDetailsService.getContactDetails(person, token);
+              ContactDetails contactDetails = contactDetailsService.getContactDetails(person);
```
Comment (ErkoRisthein) [inline src/main/java/ee/tuleva/onboarding/aml/AmlAutoChecker.java:52]: At the time when AfterTokenGrantedEvent was fired, there was no security context set up yet. That's why we needed to manually pass the token to epis. Is it now different?

File: build.gradle.kts [hunk 3]
```diff
@@ -65,14 +65,16 @@ dependencies {
...
-    implementation("com.nimbusds:nimbus-jose-jwt:9.25.6")
+    implementation("org.springframework.security:spring-security-oauth2-authorization-server:0.3.1")
+    implementation("org.springframework.security:spring-security-oauth2-client")
 
     implementation("org.springdoc:springdoc-openapi-ui:1.6.12")
     implementation("org.springdoc:springdoc-openapi-security:1.6.12")
 
+    implementation("org.glassfish.jaxb:jaxb-runtime")
+    implementation("jakarta.xml.bind:jakarta.xml.bind-api:4.0.0")
+    implementation("javax.xml.bind:jaxb-api:2.3.1")
```
Comment (ErkoRisthein) [inline build.gradle.kts:76]: Do we need the old `javax.xml` dependency? Can we replace with `jakarta.xml`?

File: src/main/java/ee/tuleva/onboarding/auth/principal/ServiceSecurityContextUtil.java [hunk 4]
```diff
@@ -0,0 +1,21 @@
...
+import org.springframework.security.core.context.SecurityContextHolder;
+
+public class ServiceSecurityContextUtil {
+
+  public static SecurityContext createServiceSecurityContext() {
+    SecurityContext context = SecurityContextHolder.createEmptyContext();
+
+    Authentication authentication =
+        new UsernamePasswordAuthenticationToken(
+            "onboarding-service", null, AuthorityUtils.createAuthorityList("ROLE_SERVICE"));
```
Comment (v3rm0n) [inline src/main/java/ee/tuleva/onboarding/auth/principal/ServiceSecurityContextUtil.java:16]: Get from Authorities files as constant

**Pull request merged**